### ISSUE TITLE
Cura 4.0.0 - Top surface low quality

### ISSUE THREAD

**Issue by amagr0** [Ultimaker/Cura#5579]
**Application Version** - CURA 4.0.0
**Platform** - Win 10 Pro, 64 bit
**Display Driver** - NVIDIA GeForce 840M

Hello,
Is there any way to avoid this:

![image](https://user-images.githubusercontent.com/12901952/55623666-a3bb0180-579b-11e9-893d-291ee6d56eb9.png)

![image](https://user-images.githubusercontent.com/12901952/55623871-1cba5900-579c-11e9-998c-ecee44d5cc65.png)

![image](https://user-images.githubusercontent.com/12901952/55623895-2d6acf00-579c-11e9-9c88-f4f0dc91cbf0.png)

![image](https://user-images.githubusercontent.com/12901952/55623923-44a9bc80-579c-11e9-9310-01d92a30f6f6.png)

![image](https://user-images.githubusercontent.com/12901952/55623948-58edb980-579c-11e9-96c1-165cc8235465.png)

The text "WHY?" will only be printed on 4th layer, but in the previous layer Cura start to do something like a tracing of the text.
This behavior increases a lot the printing time, and worst than that, decrease a lot the quality of the top surface of the 3th layer.

I think this was not always how Cura worked , but somehow this changed.
Is there any obvious reason to why this was changed?

Is amazing how complex the path is becoming to solve something that in the past was so simple.

Thanks

**Comment by smartavionics** [comment]
It's because you have top skin layers (or whatever it is called) non-zero.

**Comment by amagr0** [comment]
But if put zero the benefits of this setting will be lost, like for example, the control of the print speed of the top surface.

**Comment by smartavionics** [comment]
That's right. If the top surface has features like the text in your example, the top skin layers feature doesn't really work well.

**Comment by smartavionics** [comment]
It needs to be smarter to know that the skin layer(s) below the letters should also be treated as top skin layer rather than just underlying skin layer.

**Comment by Ghostkeeper** [comment]
Well, but then it's going to print the rest of the skin at a lower speed as well. You might as well just set the top/bottom print speed lower then.

Because these top skin lines can have different line width, speeds, etc. they will get planned separately and thus not necessarily printed in order. After all, it's completely impossible to print the lines continuously lengthwise if they change to a different line width. In your example, the "why" text on layer 3 is printed with the normal skin settings but the rest of the layer is printed with top skin.

I don't really see a way to prevent this. Trying to detect what the user intended typically causes more damage than it heals because it focusses on specific geometries and makes the whole system more complex.

**Comment by GregValiant** [comment]
I'm cleaning up these old reports.  I'll go ahead and close this.  If the problem has persisted into 5.13.0 please make out a new report.
